public/app/features/dashboard/components/PanelEditor/utils.ts

```diff
@@ -48,19 +48,38 @@ export const updateDefaultFieldConfigValue = (
       if (remove) {
         defaults.custom = omit(defaults.custom, name);
       } else {
-        defaults.custom = lodashSet({ ...defaults.custom }, name, value);
+        defaults.custom = setOptionImmutably({ ...defaults.custom }, name, value);
       }
     } else if (!remove) {
-      defaults.custom = lodashSet({ ...defaults.custom }, name, value);
+      defaults.custom = setOptionImmutably({ ...defaults.custom }, name, value);
     }
   } else if (remove) {
     defaults = omit(defaults, name);
   } else {
-    defaults = lodashSet({ ...defaults }, name, value);
+    defaults = setOptionImmutably({ ...defaults }, name, value);
   }
 
   return {
     ...config,
     defaults,
   };
 };
+
+export function setOptionImmutably<T extends object>(options: T, path: string, value: any): T {
+  const splat = path.split('.');
+  let result = { ...options };
+
+  if (splat.length === 1) {
+    (result as Record<string, any>)[path] = value;
+    return result;
+  }
+
+  let current = (result as Record<string, any>)[splat[0]];
+  if (current === undefined || current === null) {
+    current = {} as object;
+  }
+  return {
+    ...result,
+    [splat[0]]: setOptionImmutably({ ...current }, splat.slice(1).join('.'), value),
```

Comment:
`{ ...current }` is maybe unnecessary since we already do `{ ...options }` on this at the top of this fn.

it should be faster for `path` to be `string | string[]` and then use an `Array.isArray()` check internally to avoid doing the split and having to do the rejoin here. you can then simply do `pathPart = path.shift()`, and pass through this array directly into the recursive calls.